File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function(nums1, nums2) {
    let res = [...nums1, ...nums2].sort((a, b) => a - b);
    let left = 0, right = res.length - 1;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);

        if (res[left] <= res[mid]) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    let medianIndex = Math.floor((res.length - 1) / 2);
    if (res.length % 2 === 0) {
        return (res[medianIndex] + res[medianIndex + 1]) / 2;
    } else {
        return res[medianIndex];
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        for(int i=0;i<nums2.size();i++){
            nums1.push_back(nums2[i]);
        }
        sort(nums1.begin(),nums1.end());

        int left = 0;
        int right = nums1.size();

        double mid = (left+right)/2.0;

        if(nums1.size()%2==0){
            return (nums1[mid] + nums1[mid-1])/2.0;
        }else{
            return (nums1[mid]);
        }
    }
};